submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/TimesProcessor.kt

```diff
@@ -13,14 +16,25 @@ import java.time.ZoneOffset.UTC
  * - If a version of submission already exists it's creation time is used.
  */
 class TimesProcessor : SubmissionProcessor {
-
     override fun process(submission: ExtendedSubmission, context: PersistenceContext) {
         val now = OffsetDateTime.now()
         val createDate = context.getSubmission(submission.accNo)?.creationTime
-        val releaseDate = submission.releaseDate?.atOffset(UTC)
 
         submission.modificationTime = now
-        submission.releaseTime = releaseDate ?: now
         submission.creationTime = createDate ?: now
+        submission.releaseTime = getReleaseDate(submission)
+    }
+
+    private fun getReleaseDate(submission: ExtendedSubmission): OffsetDateTime {
+        var releaseDate = OffsetDateTime.now()
+        submission.releaseDate?.let {
+            try {
+                releaseDate = LocalDate.parse(it).atStartOfDay().atOffset(ZoneOffset.UTC)
+            } catch (exception: DateTimeParseException) {
+                throw InvalidDateFormatException(it)
+            }
+        }
+
+        return releaseDate
```

Comment:
what about
`submission.releaseDate?.let { parseDate(it) } ?: now`

and
```
fun parseDate(date:String): OffsetDateTime {
    return  Try.just(LocalDate.parse(it)
                    .atStartOfDay()
                    .atOffset(UTC))
                    .fold({it}, { throw InvalidDateFormatException(date) })
}
```

Reply:
👍